pkg/ring/kv/etcd/etcd.go

```diff
@@ -34,10 +40,41 @@ func (cfg *Config) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.Var((*flagext.StringSlice)(&cfg.Endpoints), prefix+"etcd.endpoints", "The etcd endpoints to connect to.")
 	f.DurationVar(&cfg.DialTimeout, prefix+"etcd.dial-timeout", 10*time.Second, "The dial timeout for the etcd connection.")
 	f.IntVar(&cfg.MaxRetries, prefix+"etcd.max-retries", 10, "The maximum number of retries to do for failed ops.")
+	// Get certs from flag params and set config TLS, if present
+	f.StringVar(&cfg.CertFile, prefix+"etcd.cert-file", "", "The TLS certificate file path")
+	f.StringVar(&cfg.KeyFile, prefix+"etcd.key-file", "", "The TLS private key file path")
+	f.StringVar(&cfg.TrustedCAFile, prefix+"etcd.client-ca-file", "", "The trusted CA file path")
+}
+
+// GetTLS sets the TLS config field with certs
+func (cfg *Config) GetTLS() (*tls.Config, error) {
+	if cfg.CertFile == "" && cfg.KeyFile == "" {
```

Comment:
@pstibrany feedback was to add another option `tls_enabled` (defaults to false) to explicitely turn it on, because on the client side (here) you may want to turn on TLS without specifying any cert/key/CA.

Reply:
Ah, makes sense. Working on this. Thanks.